packages/ketcher-react/src/script/ui/views/components/ContextMenu/ContextMenuTrigger.tsx

```diff
@@ -0,0 +1,140 @@
+/****************************************************************************
+ * Copyright 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+import { FunctionalGroup } from 'ketcher-core'
+import { useCallback } from 'react'
+import { useContextMenu } from 'react-contexify'
+import { useAppContext } from 'src/hooks'
+import Editor from 'src/script/editor'
+import { CONTEXT_MENU_ID } from './ContextMenu'
+import type { ContextMenuItemProps } from './contextMenu.types'
+
+const ContextMenuTrigger: React.FC = ({ children }) => {
+  const { getKetcherInstance } = useAppContext()
+  const { show, hideAll } = useContextMenu<ContextMenuItemProps>({
+    id: CONTEXT_MENU_ID
+  })
+
+  /**
+   * Resolve conflicts with the existing functional-group context menu
+   * Note: the following scenarios are compatible with the existing logic process,
+   *       but maybe will change after refactoring the functional-group context menu.
+   *
+   * Scenario 1
+   * Given: an expanded functional group
+   * When: the user right-clicks a bond in the f-group
+   * Then: only the functional-group context menu shows
+   *
+   * Scenario 2
+   * Given: the user has selected a bunch of things including functional groups
+   * When: the user right-clicks a selected or non-selected bond
+   * Then: only the functional-group context menu shows
+   *
+   * More details: https://github.com/epam/ketcher/pull/1896
+   */
+  const hasConflictWithFunctionalGroupMenu = useCallback(
+    (ci: any) => {
```

Comment:
If that's possible, could you, please, add types?
My bet, you have found it somewhere and there were no types as well. So, I think, it would be great to add types there.
In addition, I propose to use more clear naming for `ci` => `closestItem`. It's clear, when you know it, but for newcomers it may be unclear.